File: 2962-count-subarrays-where-max-element-appears-at-least-k-times/2962-count-subarrays-where-max-element-appears-at-least-k-times.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        int max = INT_MIN;
        for(int n : nums){
            max = n > max ? n:max;
        }
        
        long long result = 0;
        int j = 0;
        
        for(int n : nums){
            k -= (max==n);
            while(k==0){
                k += (max==nums[j++]);
            }
            result = result + j;
        }
        return result;
    }
};

File: 2962-count-subarrays-where-max-element-appears-at-least-k-times/2962-count-subarrays-where-max-element-appears-at-least-k-times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int maxElement = Integer.MIN_VALUE;
        for(int n : nums){
            maxElement = Math.max(maxElement,n);
        }
        
        int left = 0;
        int right = 0;
        int count = 0;
        long ans = 0;
        
        while(right < nums.length || left > right){
            if(nums[right] == maxElement){
                count++;
            }
            while(count >= k){
                if(nums[left] == maxElement){
                    count--;
                }
                ans = ans + nums.length - right;
                left++;
            }
            right++;
        }
        return ans;
    }
}